src/aind_dynamic_foraging_basic_analysis/metrics/session_metrics.py

```diff
@@ -0,0 +1,217 @@
+"""
+
+    Consolidated session metric tool
+    dfs = nwb_utils.create_trials_df(your_nwb)
+
+"""
+
+import re
+
+import numpy as np
+import pandas as pd
+
+from aind_dynamic_foraging_basic_analysis import compute_foraging_efficiency
+
+# Copied metrics are from process_nwbs.py in bonsai basic
+# NOTE: finished_rate_with_autowater is the same as the calculated total response rate
+LEFT, RIGHT, IGNORE = 0, 1, 2
+
+
+def session_metrics(nwb):
+    """
+    Compute all session level metrics
+
+    Includes session level metadata as a temporary organizer
+    Includes majority of metrics from process_nwbs.py
+
+    New addition: chosen_probability - average difference between the chosen probability
+    and non-chosen probability / the difference between the largest and smallest probability
+    in the session
+    """
+
+    if not hasattr(nwb, "df_trials"):
+        print("You need to compute dfs: nwb_utils.create_trials_df(nwb)")
+        return
+
+    df = nwb.df_trials.copy()
+
+    # METADATA PLACEHOLDER
+    session_start_time = nwb.session_start_time
+    session_date = session_start_time.strftime("%Y-%m-%d")
+    subject_id_from_meta = nwb.subject.subject_id
+
+    # Parse the file name for suffix
+    old_re = re.match(
+        r"(?P<subject_id>\d+)_(?P<date>\d{4}-\d{2}-\d{2})(?:_(?P<n>\d+))?\.json", nwb.session_id
+    )
+
+    if old_re is not None:
+        subject_id, session_date, nwb_suffix = old_re.groups()
+        nwb_suffix = int(nwb_suffix) if nwb_suffix is not None else 0
+    else:
+        subject_id, session_date, session_json_time = re.match(
+            r"(?P<subject_id>\d+)_(?P<date>\d{4}-\d{2}-\d{2})(?:_(?P<time>.*))\.json",
+            nwb.session_id,
+        ).groups()
+        nwb_suffix = int(session_json_time.replace("-", ""))
+
+    # Verify metadata matches
+    assert subject_id == subject_id_from_meta, (
+        f"Subject name from the metadata ({subject_id_from_meta}) does not match "
+        f"that from json name ({subject_id})!!"
+    )
+    assert session_date == session_date, (
+        f"Session date from the metadata ({session_date}) does not match "
+        f"that from json name ({session_date})!!"
+    )
+
+    # Create session index
+    session_index = pd.MultiIndex.from_tuples(
+        [(subject_id, session_date, nwb_suffix)], names=["subject_id", "session_date", "nwb_suffix"]
+    )
+
+    # Get metadata from nwb.scratch
+    meta_dict = nwb.scratch["metadata"].to_dataframe().iloc[0].to_dict()
```

Comment:
The metadata is going to be removed from scratch, so lets not use it

Reply:
Is metadata going to be moved to a different nwb module? Or should I just leave out all metrics with references to scratch?

Reply:
It will eventually be included in `nwb.lab_metadata`, but not available yet. For now, leave out anything that depends on the metadata